addon/content/modules/contact.js

```diff
@@ -36,48 +24,20 @@ var Contacts = {
 const defaultPhotoURI =
   "chrome://messenger/skin/addressbook/icons/contact-generic.png";
 
-var ContactHelpers = {
-  composeMessage(name, email, displayedFolder) {
-    let dest =
-      !name || name == email
-        ? email
-        : MailServices.headerParser.makeMimeAddress(name, email);
-    composeMessageTo(dest, displayedFolder);
-  },
-
-  showMessagesInvolving(win, name, email) {
-    let q1 = Gloda.newQuery(Gloda.NOUN_IDENTITY);
-    q1.kind("email");
-    q1.value(email);
-    q1.getCollection({
-      onItemsAdded: function _onItemsAdded(aItems, aCollection) {},
-      onItemsModified: function _onItemsModified(aItems, aCollection) {},
-      onItemsRemoved: function _onItemsRemoved(aItems, aCollection) {},
-      onQueryCompleted: function _onQueryCompleted(aCollection) {
-        if (!aCollection.items.length) {
-          return;
-        }
-
-        let q2 = Gloda.newQuery(Gloda.NOUN_MESSAGE);
-        q2.involves.apply(q2, aCollection.items);
-        q2.getCollection({
-          onItemsAdded: function _onItemsAdded(aItems, aCollection) {},
-          onItemsModified: function _onItemsModified(aItems, aCollection) {},
-          onItemsRemoved: function _onItemsRemoved(aItems, aCollection) {},
-          onQueryCompleted: function _onQueryCompleted(aCollection) {
-            const browser = BrowserSim.getBrowser();
-            let tabmail = win.document.getElementById("tabmail");
-            tabmail.openTab("glodaList", {
-              collection: aCollection,
-              title: browser.i18n.getMessage("involvingTabTitle", [name]),
-              background: false,
-            });
-          },
-        });
-      },
-    });
-  },
-};
+/*
+ * Searches a given email address in all identities and returns the corresponding identity.
+ * @param {String} anEmailAddress Email address to be searched in the identities
+ * @returns {{Boolean} isDefault, {{nsIMsgIdentity} identity} if found, otherwise undefined
+ */
+async function getIdentityForEmail(email) {
+  const browser = BrowserSim.getBrowser();
```

Comment:
Can we convert these to use the [defineLazyGetter](https://github.com/protz/thunderbird-conversations/blob/d7c365fa530f8e18a388da6ec9fdfacad08cc3be/addon/content/modules/message.js#L40-L42) in this file?

I meant to do it the other day and forgot, but should keep on top of performance.

Reply:
I don't understand... `BrowserSim` is already defined as a lazy getter... Do you mean to define `browser` in a module and then import `browser` via a lazy getter?

Reply:
Ah, I see your code link and understand.

Reply:
Eventually, this will all happen in a webextension environment where `defineLazyGetter` is unavailable. Is it really worth it to do this? It's already happening in an async function.

Reply:
Once we're in the WebExtension environment, we'll already have `browser` so we won't need to be getting it manually, so if anything having it defined globally/lazily is better.